lib/stemcell/launcher.rb

```diff
@@ -279,11 +280,16 @@ def do_launch(opts={})
       return instances
     end
 
-    def set_tags(instances=[],tags)
+    def set_tags(instances=[], tags)
       @log.info "setting tags on instance(s)"
-      instances.each do |instance|
-        instance.tags.set(tags)
+      errors = run_batch_operation(instances) do |instances|
+        begin
+          instance.tags.set(tags)
+        rescue AWS::EC2::Errors::InvalidInstanceID::NotFound => e
+          e
+        end
       end
+      check_errors(:set_tags, instances, errors)
```

Comment:
Should we map instances to their ids for the second argument here, to be consistent with `kill`?